crux_core/src/bridge/request_serde.rs

```diff
@@ -4,29 +4,35 @@ use crate::{
     Request,
 };
 
-type ResolveOnceBytes = Box<dyn FnOnce(&[u8]) + Send>;
-type ResolveManyBytes = Box<dyn Fn(&[u8]) -> Result<(), ()> + Send>;
+type ResolveOnceSerialized = Box<dyn FnOnce(&mut dyn erased_serde::Deserializer) + Send>;
+type ResolveManySerialized =
+    Box<dyn Fn(&mut dyn erased_serde::Deserializer) -> Result<(), ()> + Send>;
 
 /// A deserializing version of Resolve
 ///
-/// ResolveBytes is a separate type because lifetime elision doesn't work
+/// ResolveSerialized is a separate type because lifetime elision doesn't work
 /// through generic type arguments. We can't create a ResolveRegistry of
 /// Resolve<&[u8]> without specifying an explicit lifetime.
 /// If you see a better way around this, please open a PR.
-pub enum ResolveBytes {
+pub enum ResolveSerialized {
```

Comment:
This rename may be a step too far, as it requires a change in `crux_macros` which would otherwise not be needed, so I'm tempted to undo that.

Reply:
Unless there is a way for us to somehow specify compatibility/requirements between the two